### ISSUE TITLE
add Fontweight to fontid

### ISSUE THREAD

**Issue by Omicronlawful** [emilk/egui#3218]
It would be usefull if the font weight could be set when creating a fontid. Im not sure if there is another way, like it is solved for italics, but i would put it into fontid

**Comment by abey79** [comment]
In particular, support for variable font weight would be nice.

**Comment by emilk** [comment]
First of all: italics in epaint is a hack. We just skew the glyphs when we paint them. Ideally we should instead switch to an italic font of the same family instead.

To fully support font weights and italics we need to have better support for selecting font _families_. For example: [Inter](https://rsms.me/inter/#weights) is a font family with many different fonts for different weights and styles (Thin, Thin Italics, Bold, Bold Italics, etc).

A font family can be stored in several .ttf files, or in the same one. To know what we have loaded, we would need to read it at some sort of font meta-data (was the loaded font thin, italics, …?). I don't think `ab_glyph` supports this at the moment: https://docs.rs/ab_glyph/latest/ab_glyph/trait.Font.html

Once that all is done, then yes, `FontId` would be the right place to add weight and italics.

**Comment by abey79** [comment]
I believe there are actually two stories here:
- Support for font families (which can already be manually done)
- Support for font with variable weight(/etc) support: https://fonts.google.com/knowledge/introducing_type/introducing_variable_fonts

The latter would be very nice to have as I understand designer rely on it more and more. As it turns out, the font we're using for Rerun _is_ variable (https://fonts.google.com/specimen/Inter).

**Comment by JustFrederik** [comment]
@emilk not sure if this is helpful, but ttf parser has an example for handling metadata: https://github.com/RazrFalcon/ttf-parser/blob/master/examples/font-info.rs

**Comment by NfNitLoop** [comment]
I'd also love bold support.

FWIW, ab_glyph added support for variable fonts here: https://github.com/alexheretic/ab-glyph/pull/67

It's accessed via the [VariableFont](https://docs.rs/ab_glyph/latest/ab_glyph/trait.VariableFont.html) trait, which is enabled by default.

That unfortunately doesn't give you access to any of the font's "named instance" settings (ex: "Regular 400" from [Inter](https://fonts.google.com/specimen/Inter)), but it would at least allow folks to adjust the `wght` variable manually. 

This should also give access to "Italic" and "Slant" variables:
https://variablefonts.io/about-variable-fonts/

**Comment by lucasmerlin** [comment]
In my project I have this for bold text support:

```rs
pub static FONT_FAMILY_BOLD: &str = "bold";

pub fn font_family_bold() -> FontFamily {
  FontFamily::Name(Arc::from(FONT_FAMILY_BOLD))
}

pub trait RichTextExt {
  fn bold(self) -> Self;
}

impl RichTextExt for RichText {
  fn bold(self) -> Self {
    self.family(font_family_bold())
  }
}
pub fn font_id_heading() -> FontId {
  FontId::new(24.0, font_family_bold())
}

pub fn initialize_fonts(ctx: &Context) {
  {
    hello_egui::material_icons::initialize(ctx);

    let mut fonts = FontDefinitions::default();
    fonts.font_data.insert(
      "raleway".to_string(),
      FontData::from_static(RALEWAY_FONT_DATA).into(),
    );
    fonts.font_data.insert(
      "raleway-bold".to_string(),
      FontData::from_static(RALEWAY_BOLD_FONT_DATA).into(),
    );
    {
      let proportional = fonts.families.get_mut(&FontFamily::Proportional).unwrap();
      proportional.insert(0, "raleway".to_string());
    }
    fonts
      .families
      .insert(font_family_bold(), vec!["raleway-bold".to_string()]);

    ctx.set_fonts(fonts);

    ctx.all_styles_mut(|style| {
      *style.text_styles.get_mut(&TextStyle::Heading).unwrap() = font_id_heading()
    })
  }
}
```

Works nicely, but if course official support and support for variable fonts would be awesome! (Implementing this should hopefully be possible with parley: https://github.com/emilk/egui/pull/5784)

**Comment by emilk** [comment]
I believe this issue will be closed to solved (if not fully solved) by:
* https://github.com/emilk/egui/pull/5784

**Comment by NfNitLoop** [comment]
Ah cool, I saw that PR but hadn't read much further than 
> This is very WIP and currently a proof-of-concept

So the goal is to merge that? I'm a bit worried by:
> There is no guarantee that I'll finish this PR; if you ping me and I respond with something like "sorry, I'll get around to it next week" a few times in a row, feel free to take over it yourself.

Anything we can do to help?
